README.md

```diff
@@ -1,20 +1,453 @@
+
+The opposite of mutable is immutable. Immutable Objects can't be modified. There are two methods `Object.seal` and `Object.freeze` that prevent properties from being added to an Object.
+
+#### Object.seal
+
+When `Object.seal` is called on an Object, that Object can no longer have new properties added to it. Pre-existing properties can be modified as long as their property descriptor set `writable` to true. This doesn't make Objects immutable because we can still modify existing properties.
+
+```js
+var pikachu = {
+	name: "Pikachu"
+};
+Object.seal(pikachu);
+pikachu.level = 100; // since this Object has been sealed, this new property will not be set
+pikachu.name = "Pika"; // pikachu.name will be modified since this propery existed before the seal and is writable
+```
+
+#### Object.freeze
+
+Similar to `Object.seal`, when `Object.freeze` is called on an Object, that Object can no longer have new properties added to it. In addition to this, existing properties can't be modified, despite their property descriptors. This makes an Object immutable.
+
+```js
+var pikachu = {
+	name: "Pikachu"
+};
+Object.freeze(pikachu);
+pikachu.level = 100; // since this Object has been frozen, this new property will not be set
+pikachu.name = "Pika"; // pikachu.name will not be modified since this propery is frozen
+```
+
+## Accessing Object Properties
+
+Earlier we noted how there were two ways of assigning a property to an Object that resembled variable assignment. We can access these assigned properties using a similar approach. We've actually seen this numerous times throghout the lesson:
+
+#### Dot Notation
+
+Just as we set a property using dot notation, we can retrieve a property by leaving out the right hand side.
+
+```js
+pikachu.name // returns "Pikachu"
+```
+
+#### Key Indexes
+
+Again, same as adding a property, except we leave out the right hand side.
+
+```js
+pikachu["name"] // returns "Pikachu"
+```
+
+If you try to access a property that doesn't exist, it will return `undefined`
+
+```js
+pikachu["randomProperty"] // returns undefined
+pikachu.randomProperty // returns undefined
+```
+
+If you're not sure that the Object is defined, its always good to check, otherwise an error will be thrown if you try to access a property on it.
+
+```js
+var bulbasaur;
+bulbasaur.name // this will throw a TypeError because bulbasaur it just an empty variable that was never defined as an Object
+```
+Most JavaScript engines will throw an error along the lines of "Uncaught TypeError: Cannot read property 'name' of undefined"
+
+## Removing Object Properties
+
+Sometimes you'll want to remove a property from an Object when its no longer necessary. This can help free up memory by removing references to variables that are no longer needed. Since JavaScript is a memory managed language theres no guarantee exactly when memory will be freed, but you can at least ensure that it will happen some time in the future by removing all unused references.
```

Comment:
I think this section is great, but we can probably cut the part about memory management -- it's beyond the scope of what students will need to practice in the near term (although at some point they should understand it, for sure).